qiskit/pulse/library/symbolic_pulses.py

```diff
@@ -547,8 +547,11 @@ def __eq__(self, other: "SymbolicPulse") -> bool:
         if self._envelope != other._envelope:
             return False
 
-        if self.parameters != other.parameters:
-            return False
+        # SymbolicPulse requires parameters to be identical, but subclasses of SymbolicPulse
+        # Have different conditions.
+        if self.__class__ == SymbolicPulse:
```

Comment:
This looks strange. This implements a part of logic of subclass in the superclass. This is not standard python inheritance. Since equality here is very simple, I wouldn't say it's a boilerplate code if you duplicate the rest of logic in the subclass.

Reply:
I think you are right. I didn't want to duplicate the code, but it came out strange. I'll change it.